src/mongo/unittest/unittest.cpp

```diff
@@ -386,6 +447,150 @@ int Suite::run(const std::vector<std::string>& suites, const std::string& filter
+        results.push_back(s->run(filter, runsPerTest));
+    }
+
+    log() << "**************************************************" << std::endl;
+
+    int rc = 0;
+
+    int tests = 0;
+    int asserts = 0;
+    int millis = 0;
+
+    Result totals("TOTALS");
+    std::vector<std::string> failedSuites;
+
+    Result::cur = NULL;
+    for (std::vector<Result*>::iterator i = results.begin(); i != results.end(); i++) {
+        Result* r = *i;
+        log() << r->toString();
+        if (abs(r->rc()) > abs(rc))
+            rc = r->rc();
+
+        tests += r->_tests;
+        if (!r->_fails.empty()) {
+            failedSuites.push_back(r->toString());
+            for (std::vector<std::string>::const_iterator j = r->_fails.begin();
+                 j != r->_fails.end();
+                 j++) {
+                const std::string& s = (*j);
+                totals._fails.push_back(r->_name + "/" + s);
+            }
+        }
+        asserts += r->_asserts;
+        millis += r->_millis;
+
+        delete r;
+    }
+
+    totals._tests = tests;
+    totals._asserts = asserts;
+    totals._millis = millis;
+
+    log() << totals.toString();  // includes endl
+
+    // summary
+    if (!totals._fails.empty()) {
+        log() << "Failing tests:" << std::endl;
+        for (std::vector<std::string>::const_iterator i = totals._fails.begin();
+             i != totals._fails.end();
+             i++) {
+            const std::string& s = (*i);
+            log() << "\t " << s << " Failed";
+        }
+        log() << "FAILURE - " << totals._fails.size() << " tests in " << failedSuites.size()
+              << " suites failed";
+    } else {
+        log() << "SUCCESS - All tests in all suites passed";
+    }
+
+    // Start building the JSON output using BSONObjBuilder
+    BSONObjBuilder b;
+
+    // TODO: Get the start and end time
+    b.append("start", "PLACEHOLDER");
+    b.append("end", "PLACEHOLDER");
+
+    // No errors
+    BSONArrayBuilder errorsList(b.subarrayStart("errors"));
+    errorsList.doneFast();
+
+    BSONArrayBuilder resultsList(b.subarrayStart("results"));
+    // Construct the results list by pulling the data from the static BenchmarkStats struct
+    for (BenchmarkStats stat : statsList) {
+        BSONObjBuilder statObj;
+        statObj.append("name", stat.name);
+        BSONObjBuilder benchResults(statObj.subobjStart("results"));
+
+        BSONObjBuilder one(benchResults.subobjStart("1"));
+        one.append("ops_per_sec", stat.ops_per_sec);
+        one.append("ops_per_sec_values", stat.ops_per_sec_vals);
+        one.append("error_values", stat.error_values);
+        one.doneFast();
+
+        benchResults.doneFast();
+
+        resultsList.append(statObj.done());
+    }
+    resultsList.doneFast();
+
+    // write to outFile or cout
+    std::string json = b.done().jsonString();
+    if (benchmarkGlobalParams.outFileName == "") {
+        std::cout << json << std::endl;
```

Comment:
If you are going to do this, you should set the pretty arg to 1 in jsonString() so the output is readable.